src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -902,70 +901,26 @@ where
             let mut buffer = Vec::new();
 
-            let mut persist_clients = persist_clients.lock().await;
-            let persist_client = persist_clients
-                .open(storage_metadata.persist_location.clone())
-                .await
-                .expect("error creating persist client for Healthchecker");
-            drop(persist_clients);
-
-            let client_ref = &persist_client;
-
-            let write_to_persist = |row: Row, timestamp: Timestamp| async move {
-                let status_shard = match storage_metadata.status_shard {
-                    Some(s) => s,
-                    None => return,
-                };
-                let handle = client_ref
-                    .open_writer(status_shard)
-                    .await;
-
-                let mut handle = match handle {
-                    Ok(stuff) => stuff,
-                    Err(e) => {
-                        panic!("Invalid usage of the persist client for source {source_id} status history shard: {e:?}");
-                    }
-                };
-
-                let mut recent_upper = handle.upper().clone();
-                let mut append_ts = timestamp;
-                'retry_loop: loop {
-                    // We don't actually care so much about the timestamp we append at; it's best-effort.
-                    // Ensure that the append timestamp is not less than the current upper, and the new upper
-                    // is past that timestamp. (Unless we've advanced to the empty antichain, but in
-                    // that case we're already in trouble.)
-                    for t in recent_upper.elements() {
-                        append_ts.join_assign(t);
-                    }
-                    let mut new_upper = Antichain::from_elem(append_ts.step_forward());
-                    new_upper.join_assign(&recent_upper);
-
-                    let updates = vec![
-                        ((SourceData(Ok(row.clone())), ()), append_ts, 1i64)
-                    ];
-                    let cas_result = handle.compare_and_append(updates, recent_upper.clone(), new_upper).await;
-                    match cas_result {
-                        Ok(Ok(Ok(()))) => break 'retry_loop,
-                        Ok(Ok(Err(Upper(upper)))) => {
-                            recent_upper = upper;
-                        }
-                        Ok(Err(e)) => {
-                            panic!("Invalid usage of the persist client for source {source_id} status history shard: {e:?}");
-                        }
-                        Err(e) => {
-                            trace!("compare_and_append in update_status failed: {e}");
-                        }
-                    }
-                }
+            let status_shard = if let Some(status_shard) = storage_metadata.status_shard {
+                status_shard
+            } else {
+                // If there's no status shard, this operator has no work to do.
```

Comment:
I'd like to keep the logging for now, if that's possible. Maybe we can wrap both the `write_to_persist` calls in an `if let`?

Reply:
makes sense -- no more using new `if let else` syntax!

Reply:
I restored the logging for sources but didn't _add_ logging to sinks when there's no status shard since sinks are set up a little differently than sources -- but could do in a follow-up if you feel strongly?